crates/bevy_reflect/src/serde/de.rs

```diff
@@ -210,6 +211,13 @@ impl<'de> Visitor<'de> for U32Visitor {
     }
 }
 
+/// Helper struct for deserializing strings without allocating (when possible).
+///
+/// Based on [this comment](https://github.com/bevyengine/bevy/pull/6894#discussion_r1045069010).
+#[derive(Deserialize)]
+#[serde(transparent)]
+struct BorrowableCowStr<'a>(#[serde(borrow)] Cow<'a, str>);
```

Comment:
Since this is used in two different crates, should this construct be moved to something like `bevy_utils`? Or is it fine to keep this duplication since it's only a few lines?

Reply:
i think the behaviour is short and niche enough to leave as is in this pr, but we should probably keep an eye on it. three uses is where i'd probably stick it into a shared crate.

Reply:
Hm, should this be something that is owned by `bevy_reflect::de` and depended on by `bevy_scene`? I feel like `bevy_reflect` shouldn't really have that responsibility, right?

I can export it if that's preferred.